sqlite_query.c

```diff
@@ -21,63 +21,82 @@
 #include "utils/builtins.h"
 #include "utils/lsyscache.h"
 
+#include "nodes/makefuncs.h"
+#include "catalog/pg_type.h"
+#include "parser/parse_type.h"
+
 static int32
-			sqlite_from_pgtyp(Oid pgtyp);
+			sqlite_affinity_eqv_to_pgtype(Oid pgtyp);
+static const char*
+			sqlite_datatype(int t);
 
 /*
  * convert_sqlite_to_pg: Convert Sqlite data into PostgreSQL's compatible data types
  */
-Datum
-sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_colid, AttInMetadata *attinmeta, AttrNumber attnum)
+NullableDatum
+sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_colid, AttInMetadata *attinmeta, AttrNumber attnum, int sqlite_value_affinity, int AffinityBehaviourFlags)
 {
 	Datum		value_datum = 0;
 	char	   *valstr = NULL;
-	int			col_type = sqlite3_column_type(stmt, stmt_colid);
-	int			sqlite_type = sqlite_from_pgtyp(pgtyp);
+	int			affinity_for_pg_column = sqlite_affinity_eqv_to_pgtype(pgtyp);
+	int		 	value_byte_size_blob_or_utf8 = sqlite3_column_bytes(stmt, stmt_colid); // Compute always, void text and void BLOB will be special cases
 
-	if (sqlite_type != col_type && col_type == SQLITE3_TEXT)
-		elog(ERROR, "invalid input syntax for type =%d, column type =%d", sqlite_type, col_type);
+	if (affinity_for_pg_column != sqlite_value_affinity && sqlite_value_affinity == SQLITE3_TEXT)
+	{
+		/* TODO: human readable error message based on this code
```

Comment:
I think the error message format or error message example is enough, not need to add the code here.

Reply:
Really. I have reduced the comment.

Reply:
Thank you.